014-single-codebase.md

```diff
@@ -0,0 +1,68 @@
+## Problem
+
+We rebuilt our old [Drupal 7 monolith](https://github.com/dosomething/website-2013) into a suite of services – [Northstar](https://github.com/DoSomething/northstar), [Rogue](https://github.com/DoSomething/rogue), [Phoenix](https://github.com/DoSomething/phoenix-next), etc. – between 2013-2019. We avoided [the quagmire of a total rewrite](https://www.joelonsoftware.com/2000/04/06/things-you-should-never-do-part-i/) by extracting functionality [piece-by-piece](https://martinfowler.com/bliki/StranglerFigApplication.html) & have been generally happy with what we've built since then.
+
+However, breaking our monolith into a service-oriented architecture introduced new issues of its own:
+
+- **Blurred boundaries:** The [bounded contexts](https://martinfowler.com/bliki/BoundedContext.html) for each of our services (especially our identity & activity APIs) can be blurry. For example, is joining a club part of a user's identity or an action that they take? We're often forced to make this decision early in a feature's development process, before the technical work has fully taken shape.
+
+- **Technical overhead:** Logic that straddles multiple domains is also inevitably more complicated, requiring network requests for what could otherwise be a function call or `JOIN` statement. It makes traditionally simple logic into distributed systems problems where we have to decide whether we'll trade consistency for performance or vice versa.
+
+- **Testing limitations:** We know we get the best bang for our buck from [high-level integration tests](https://github.com/DoSomething/rfcs/blob/master/004-testing.md), but separating into services forces us to use mocking extensively because the majority of our application (e.g. users in Rogue, or users & posts in Phoenix) is always an external dependency in any given test! This places a large burden on our intra-service contracts.
+
+- **Knowledge silos:** Finally, this separation reinforces silos (e.g. "you have to talk to Dave about that app" or "that's X application's responsibility, not ours"). We have a smaller team today than when we first started splitting out into a service oriented architecture, and it's important that we can all understand & comfortably work on the whole system.
+
+
+
+## Proposal
+
+We should consolidate our backend services (Northstar, Rogue, and Chompy) into **a single monolithic Laravel application.**
+
+- This will **simplify architectural decisions** & enable us to iterate faster. Instead of sorting new functionality into strict bounded contexts up front, we'll be able to simply place new entities into a single backend context. Architectural decisions will be less binding, as it's easier to refactor within a single codebase than across network boundaries.
+
+- It will also **simplify adding behavior that straddles contexts**, like clubs, by replacing network requests with function calls. We'll also be able to ensure better consistency by using more transactional validation.
```

Comment:
This bit definitely is something where we've had long discussions about trade-offs before and I like the benefit we get here!